### PR TITLE
Add functional tests for handler methods

### LINKED ISSUE #5841: Add Functional Tests: Handler Methods

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #5841.

### CODE REVIEW

**Comment by pranavkm** [comment]
Need tests with handlers with form actions, `OnGetCustomer`, async handler methods (`OnPostAsync`) and a combination of both `OnPostDeleteCustomerAsync`.

**Comment by rynowak** [comment]
I'd also want to see different signatures for the handler method and handler methods defined on the page vs a page model.

These different return types all get different treatment by the runtime:
`Task`
`Task<IActionResult>`
`Task<T>` where `T` is some class that implements `IActionResult`
`void`
`IActionResult`
`T` where `T` is some class that implements `IActionResult`

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/WebSites/RazorPagesWebSite/CustomActionResultModel.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        public async Task OnGetCustomer()
+        {
+            await Task.Delay(1);
+            MethodName = nameof(OnGetCustomer);
+        }
+
+        public async Task OnGetViewCustomerAsync()
+        {
+            await Task.Delay(1);
```
Comment (pranavkm) [inline test/WebSites/RazorPagesWebSite/CustomActionResultModel.cs]: Eww, just have these return `TaskCache.CompletedTask`. Shouldn't use delays unless you need them.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/WebSites/RazorPagesWebSite/PageCustomActionResultPage.cshtml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿@page "{formaction?}"
```
Comment (pranavkm) [inline test/WebSites/RazorPagesWebSite/PageCustomActionResultPage.cshtml]: `HandlerTestPage`? It does more than just `CustomActionResult`

Comment (pranavkm) [inline test/WebSites/RazorPagesWebSite/PageCustomActionResultPage.cshtml]: Similarly, for the model.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 3]
```diff
@@ -20,6 +20,187 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
...
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var content = await response.Content.ReadAsStringAsync();
+            Assert.StartsWith("Method: OnGetViewCustomerAsync", content.Trim());
+        }
+
+        [Fact]
+        public async Task Page_Handler_ReturnTypeImplementsIActionResult()
+        {
+            // Arrange
+            var getRequest = new HttpRequestMessage(HttpMethod.Get, "http://localhost/PageCustomActionResultPage?message=message");
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Do we use `message`?

**Pull request merged**